polaris-react/src/tokens/token-groups/beta-tokens.json

```diff
@@ -0,0 +1,7 @@
+{
+  "animation-name-bounce": "p-keyframes-bounce",
+  "animation-name-fade-in": "p-keyframes-fade-in",
+  "animation-name-fill-up": "p-keyframes-fill-up",
+  "animation-name-pulse": "p-keyframes-pulse",
+  "animation-name-spin": "p-keyframes-spin"
```

Comment:
`beta-tokens.json` is an idea @lgriffee, @BPScott, and I came up with to allow adding new (experimental) tokens, that are by nature publicly accessible, yet internally marked `beta`. This should give us the freedom to make updates without introducing breaking changes.

If it helps we can additionally prepend `beta-*` to each token name.

In the case of these `animation-name-*` tokens, I am interested in experimenting with the [Open Props approach to animation tokens](https://github.com/argyleink/open-props/blob/40ae491289f38939ce89af05f44eb0249e874db6/src/props.animations.css#L2-L21) where we'd expose custom properties for the `animation:` shorthand. These `animation-name-*` tokens help accomplish our immediate goal to improve `@keyframes` coverage while laying the ground work for introducing `animation:` shorthand custom properties e.g.
```json5
// token-groups/motion.json
{
  "animation-name-bounce": "p-keyframes-bounce",
  "animation-bounce": "var(--p-animation-name-bounce) var(--p-duration-200) var(--p-ease)",
 }
```
```css
/* MyComponent.scss */
.usage {
  animation: var(--p-animation-bounce);
  animation-duration: optional override;
}
```

Reply:
I am not a huge fan of having beta tokens. Why would we not ship the values and iterate upon them?

Reply:
I also don't like the two seperate files for animation and keyframe values. Is there any reason we wouldn't consolidate these? This PR seems to add a lot of complexity. What is the simplest way we could do this?

Reply:
> Why would we not ship the values and iterate upon them?

How can consumers tell that tokens are stable and are thus safe to use in their apps? At the moment all tokens are considered available for use in consuming apps and thus they're part of a stable api - removing a token is a breaking change that warrants a major version release. Thus polaris-react can't iterate when releasing minor versions without leaving a trail of "deprecated, we don't use this anymore as we found a better approach" tokens in its wake till the next major version, or by having regular major version releases (which is bad because it leads to update fatigue in consuming apps - people get sick of chasing polaris). The alternative here is that the whole of polaris goes into a long running beta but that's bad for other reasons (e.g. keeping main and the beta branch in sync).


By marking a subset of tokens as "beta" then we explicitly denounce that stability guarantee - polaris would be free to iterate and remove them at any point in time as they're explictly not recommended for usage within consuming apps. Once we're happy with an approach then tokens can be promoted up to the stable set where the usual "please use this, it won't go away except in a major version" constraints exist.

Thus we end up with 3 tiers of token:

- beta: Unstable, we're still working out if we like this approach. Don't use this as it might go away unexpectedly. May be promoted up to stable
- stable: Stable, use this in your consuming apps. It won't go away
- deprecated: Stable, but shall go away in the next major version. We don't use this internally. Please migrate away from this in external apps.

Reply:
If everything is stable then how do you propose that polaris "iterate[s] upon them"? Shall we consider it acceptable that we'll probably end up with a handful of tokens that won't be used within the polaris-react codebase, but consumers may use (as there is no warning not to use them), and then we have to migrate folks away come the next major version when we remove tokens?